Content.Server/Body/Systems/BloodstreamSystem.cs

```diff
@@ -375,4 +375,20 @@ public void SpillAllSolutions(EntityUid uid, BloodstreamComponent? component = n
             }
         }
     }
+
+    /// <summary>
+    ///     Change what's in someone's blood, on the fly. Will also refill their blood, I guess.
+    /// </summary>
+    public void ChangeBloodReagent(EntityUid uid, string reagent, BloodstreamComponent? component = null)
+    {
+        if (!Resolve(uid, ref component, false))
+            return;
+
+        if(reagent == component.BloodReagent)
+            return;
+
+        component.BloodReagent = reagent;
+        component.BloodSolution.RemoveAllSolution();
+        _solutionContainerSystem.TryAddReagent(uid, component.BloodSolution, component.BloodReagent, component.BloodMaxVolume, out _);
```

Comment:
Zombies don't take bloodloss damage anyways so I didn't see it being an issue if they had the same amount of blood or not. I'll still fix it since it's really not that hard.